src/tensors/factorizations/matrixalgebrakit.jl

```diff
@@ -0,0 +1,528 @@
+
+function check_input(::typeof(lq_full!), t::AbstractTensorMap, (L, Q)::_T_LQ)
+    # scalartype checks
+    @check_scalar L t
+    @check_scalar Q t
+
+    # space checks
+    V_Q = fuse(domain(t))
+    @check_space(L, codomain(t) ← V_Q)
+    @check_space(Q, V_Q ← domain(t))
+
+    return nothing
+end
+
+function check_input(::typeof(lq_compact!), t::AbstractTensorMap, (L, Q)::_T_LQ)
+    # scalartype checks
+    @check_scalar L t
+    @check_scalar Q t
+
+    # space checks
+    V_Q = infimum(fuse(codomain(t)), fuse(domain(t)))
+    @check_space(L, codomain(t) ← V_Q)
+    @check_space(Q, V_Q ← domain(t))
+
+    return nothing
+end
+
+function check_input(::typeof(lq_null!), t::AbstractTensorMap, N)
+    # scalartype checks
+    @check_scalar N t
+
+    # space checks
+    V_Q = infimum(fuse(codomain(t)), fuse(domain(t)))
+    V_N = ⊖(fuse(domain(t)), V_Q)
+    @check_space(N, V_N ← domain(t))
+
+    return nothing
+end
+
+function initialize_output(::typeof(lq_full!), t::AbstractTensorMap, ::AbstractAlgorithm)
+    V_Q = fuse(domain(t))
+    L = similar(t, codomain(t) ← V_Q)
+    Q = similar(t, V_Q ← domain(t))
+    return L, Q
+end
+
+function initialize_output(::typeof(lq_compact!), t::AbstractTensorMap, ::AbstractAlgorithm)
+    V_Q = infimum(fuse(codomain(t)), fuse(domain(t)))
+    L = similar(t, codomain(t) ← V_Q)
+    Q = similar(t, V_Q ← domain(t))
+    return L, Q
+end
+
+function initialize_output(::typeof(lq_null!), t::AbstractTensorMap, ::AbstractAlgorithm)
+    V_Q = infimum(fuse(codomain(t)), fuse(domain(t)))
+    V_N = ⊖(fuse(domain(t)), V_Q)
+    N = similar(t, V_N ← domain(t))
+    return N
+end
+
+# Polar decomposition
+# -------------------
+const _T_WP = Tuple{<:AbstractTensorMap,<:AbstractTensorMap}
+const _T_PWᴴ = Tuple{<:AbstractTensorMap,<:AbstractTensorMap}
+using MatrixAlgebraKit: PolarViaSVD
+
+function check_input(::typeof(left_polar!), t::AbstractTensorMap, (W, P)::_T_WP)
+    codomain(t) ≿ domain(t) ||
+        throw(ArgumentError("Polar decomposition requires `codomain(t) ≿ domain(t)`"))
+
+    # scalartype checks
+    @check_scalar W t
+    @check_scalar P t
+
+    # space checks
+    @check_space(W, space(t))
+    @check_space(P, domain(t) ← domain(t))
+
+    return nothing
+end
+
+function check_input(::typeof(left_orth_polar!), t::AbstractTensorMap, (W, P)::_T_WP)
+    codomain(t) ≿ domain(t) ||
+        throw(ArgumentError("Polar decomposition requires `codomain(t) ≿ domain(t)`"))
+
+    # scalartype checks
+    @check_scalar W t
+    @check_scalar P t
+
+    # space checks
+    VW = fuse(domain(t))
+    @check_space(W, codomain(t) ← VW)
+    @check_space(P, VW ← domain(t))
+
+    return nothing
+end
+
+function initialize_output(::typeof(left_polar!), t::AbstractTensorMap, ::AbstractAlgorithm)
+    W = similar(t, space(t))
+    P = similar(t, domain(t) ← domain(t))
+    return W, P
+end
+
+function check_input(::typeof(right_polar!), t::AbstractTensorMap, (P, Wᴴ)::_T_PWᴴ)
+    domain(t) ≿ codomain(t) ||
```

Comment:
Is there a specific preference for writing the checks using `≿` versus `≾` ?

I know this is nitpicking, but I would either prefer consistent use of `≾` , or alternatively, consistent use of `codomain(t)` as first argument and `domain(t)` as second, with varying operator.